client/my-sites/people/main.jsx

```diff
@@ -33,9 +33,16 @@ class People extends React.Component {
 			case 'team':
 				return <TeamList site={ site } search={ search } />;
 			case 'followers':
-				return <FollowersList site={ site } />;
+				return <FollowersList key={ `people-followers-${ site.ID }` } site={ site } />;
```

Comment:
This makes sure we get a fresh list component (and don't mess with state updates) in case a user changes their site (opposed to having custom logic in the component itself which listens for changes to `site.id`). I'm not sure though whether this counts as _hacky_ way to accomplish this.

Reply:
🤔  If the user changes their site, `site` prop will change and a new `site` prop will be passed to `FollowersList` anyway. So I'm not sure I understand - can you elaborate?

Reply:
We handle internal state (`page`) which won't reset if a user changes their site so if a user fetched a few pages on one site and changes to another site it'd reuse that `page` prop which we don't want. Instead of resetting that manually inside the component we just mount a fresh instance which will then get the initial state. I'm not sure if that's a good approach but I've seen it in a few places and it _feels_ a little bit cleaner.

looping in @jsnajdr for additional feedback on this approach

Reply:
This actually makes a lot of sense. Thanks for elaborating, I think it's a solid approach and we've done it before.

If could further nitpick, let's add a comment explaining why we're adding a `key` prop.

Reply:
added a comment via 3456f86